gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/UserServiceImpl.java

```diff
@@ -313,10 +331,22 @@ public Single<User> update(String userId, User user, String idp, String baseUrl,
                                                                     return userProvider.create(UserMapper.convert(userToUpdate));
                                                                 } else {
                                                                     if (isNullOrEmpty(userToUpdate.getPassword())) {
-                                                                        return userProvider.update(userToUpdate.getExternalId(), UserMapper.convert(userToUpdate));
+                                                                        return createPasswordHistory(
```

Comment:
Maybe we should check the behaviour here.
If `isNullOrEmpty(userToUpdate.getPassword())` return true, why should we call createPasswordHistory instead of just updating the user ?